WaterJug.py

```diff
@@ -4,101 +4,102 @@
 
 class WaterJug:
 
-    def __init__(self,fileptr):
-        self.num=int(fileptr.readline())
+    def __init__(self,fileptr): #constructor
```

Comment:
I believe these kinds of comments are use less. Some one who knows python, will be able to understand what this function is supposed to do. Putting a comment like this just a distraction.